nucleus/slice.py

```diff
@@ -17,6 +19,30 @@
 )
 
 
+class SliceBuilderMethods(str, Enum):
+    RANDOM = "Random"
+    UNIQUENESS = "Uniqueness"
+
+
+@dataclass
+class SliceBuilderFilterAutotag:
+    autotag_id: str
+    score_range: List[int]
+
+    def __post_init__(self):
+        warn_msg = f"Autotag score range must be within [-1, 1]. But got {self.score_range}."
+        assert len(self.score_range) == 2, warn_msg
+        assert (
+            min(self.score_range) >= -1 and max(self.score_range) <= 1
+        ), warn_msg
+
+
+@dataclass
+class SliceBuilderFilters:
+    slice_id: Optional[str] = None
+    autotag: Optional[SliceBuilderFilterAutotag] = None
+
```

Comment:
I was indecisive to whether go this approach, or just simplify everything into one named-tuple, ie:
```
Filter = namedtuple('Filter', ['slice_id', 'autotag_id', 'range'])
```

Reply:
You made the right choice. I used a named tuple for my test filters and I regret that horribly. It is nigh impossible to change without doing a massive backfill. Much better to have named fields.